pkg/core/native/native_neo.go

```diff
@@ -670,29 +648,30 @@ func makeVoterKey(pub []byte, prealloc ...[]byte) []byte {
 
 // CalculateBonus calculates amount of gas generated for holding value NEO from start to end block
 // and having voted for active committee member.
-func (n *NEO) CalculateBonus(d *dao.Simple, acc util.Uint160, end uint32) (*big.Int, error) {
+func (n *NEO) CalculateBonus(ic *interop.Context, acc util.Uint160) (*big.Int, error) {
 	key := makeAccountKey(acc)
-	si := d.GetStorageItem(n.ID, key)
+	si := ic.DAO.GetStorageItem(n.ID, key)
 	if si == nil {
 		return nil, storage.ErrKeyNotFound
 	}
 	st, err := state.NEOBalanceFromBytes(si)
 	if err != nil {
 		return nil, err
 	}
-	return n.calculateBonus(d, st.VoteTo, &st.Balance, st.BalanceHeight, end)
+	return n.calculateBonus(ic, st)
 }
 
-func (n *NEO) calculateBonus(d *dao.Simple, vote *keys.PublicKey, value *big.Int, start, end uint32) (*big.Int, error) {
-	r, err := n.CalculateNEOHolderReward(d, value, start, end)
-	if err != nil || vote == nil {
+func (n *NEO) calculateBonus(ic *interop.Context, acc *state.NEOBalance) (*big.Int, error) {
+	end := ic.BlockHeight() + 1
```

Comment:
This is the only place you need a proper `ic`, so I'd rather keep this function as `func (n *NEO) calculateBonus(d *dao.Simple, acc *state.NEOBalance, end uint32)`.

Reply:
Done